src/Bicep.Core/TypeSystem/Az/AzResourceTypeLoaderFactory.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Bicep.Types.Az;
+using Bicep.Core.Syntax;
+using System.IO;
+using Bicep.Core.Registry.Oci;
+using Newtonsoft.Json;
+using Bicep.Core.Features;
+
+namespace Bicep.Core.TypeSystem.Az
+{
+    public class AzResourceTypeLoaderFactory : IAzResourceTypeLoaderFactory
+    {
+        private readonly IFeatureProviderFactory featureProviderFactory;
+
+        private Dictionary<string, IAzResourceTypeLoader> resourceTypeLoaders = new(){
+            {"builtin", new AzResourceTypeLoader(new AzTypeLoader())},
+        };
+
+        public AzResourceTypeLoaderFactory(IFeatureProviderFactory featureProviderFactory)
+        {
+            this.featureProviderFactory = featureProviderFactory;
+        }
+
+        public IAzResourceTypeLoader? GetResourceTypeLoader(ImportDeclarationSyntax? ids, IFeatureProvider features)
+        {
+            if (!features.DynamicTypeLoading || ids is null)
+            {
+                return resourceTypeLoaders["builtin"];
+            }
+
+            var azProviderDir = Path.Combine(features.CacheRootDirectory, "br", "mcr.microsoft.com", @"bicep$providers$az", ids.Specification.Version);
+            var ociManifestPath = Path.Combine(azProviderDir, "manifest.json");
```

Comment:
>"manifest.json" [](http://example.com/codeflow?start=62&length=15)

We named it `manifest` in the cache without the extension. There's no reason to deviate from that for providers.

Reply:
Thanks for renaming.